smithy-model/src/main/java/software/amazon/smithy/model/traits/ExamplesTrait.java

```diff
@@ -133,8 +133,8 @@ public ObjectNode getInput() {
         /**
          * @return Gets the output object.
          */
-        public ObjectNode getOutput() {
-            return output;
+        public Optional<ObjectNode> getOutput() {
```

Comment:
A bit of a breaking change... but maybe using this trait in code is fringe enough to not matter

Reply:
I thought about keeping it as is and checking if `example.getOutput().isEmpty()` to check if it was defined, but it opened the possibility of the example below being valid, which I think shouldn't be, if it's present it should conform, even if empty:

```
$version: "2.0"

namespace example.hello

@examples([
    {
        title: "Unhappy path",
        output: {},
        error: {
            shapeId: YouShallNotPass
            content: {
                "message": "Go back to the Shadow"
            }
        }
    }
])
operation Hello {
    output: Greeting,
    errors: [YouShallNotPass]
}

structure Greeting {
    @required
    message: String
}

@error("client")
structure YouShallNotPass {
    @required
    message: String
}
```